ClickHouse.Client/Copy/ClickHouseBulkCopy.cs

```diff
@@ -165,13 +169,24 @@ private async Task PushBatch(ICollection<object[]> rows, ClickHouseType[] column
                 }
 
                 using var writer = new ExtendedBinaryWriter(gzipStream);
-                foreach (var row in rows)
+                object[] excp_rows = null;
+
+                try
                 {
-                    for (var i = 0; i < row.Length; i++)
+                    foreach (var row in rows)
                     {
-                        columnTypes[i].Write(writer, row[i]);
+                        excp_rows = row;
+                        for (var i = 0; i < row.Length; i++)
+                        {
+                            columnTypes[i].Write(writer, row[i]);
+                        }
                     }
                 }
+                catch (Exception ex)
+                {
+                    Console.WriteLine(ex.ToString());
+                    OnBulkCopyException(this, excp_rows);
```

Comment:
This calls for a separate exception type (e.g. `ClickHouseSerializationException`) which has a field with the failing row